lib/Extension/PhpSpec/Provider/ObjectBehaviorMemberProvider.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Phpactor\Extension\PhpSpec\Provider;
+
+use Phpactor\WorseReflection\Core\ClassName;
+use Phpactor\WorseReflection\Core\Exception\NotFound;
+use Phpactor\WorseReflection\Core\Reflection\Collection\ReflectionMemberCollection;
+use Phpactor\WorseReflection\Core\Reflection\ReflectionClassLike;
+use Phpactor\WorseReflection\Core\ServiceLocator;
+use Phpactor\WorseReflection\Core\Type\ClassType;
+use Phpactor\WorseReflection\Core\Virtual\Collection\VirtualReflectionMemberCollection;
+use Phpactor\WorseReflection\Core\Virtual\Collection\VirtualReflectionMethodCollection;
+use Phpactor\WorseReflection\Core\Virtual\ReflectionMemberProvider;
+use Phpactor\WorseReflection\Core\Virtual\VirtualReflectionMethod;
+
+class ObjectBehaviorMemberProvider implements ReflectionMemberProvider
+{
+    private const OBJECT_BEHAVIOR_CLASS = '\PhpSpec\ObjectBehavior';
+    private const SUBJECT_CLASS = '\PhpSpec\Wrapper\Subject';
+    
+    private string $specPrefix;
+
+    public function __construct(string $specPrefix = 'spec')
+    {
+        $this->specPrefix = $specPrefix;
+    }
+
+    public function provideMembers(
+        ServiceLocator $serviceLocator,
+        ReflectionClassLike $class
+    ): ReflectionMemberCollection {
+        $subjectClassName = explode('\\', $class->name()->namespace());
+
+        if (false === $this->isSpecCandidate($class, $subjectClassName)) {
+            return VirtualReflectionMemberCollection::fromMembers([]);
+        }
+
+        array_shift($subjectClassName);
+        $subjectClassName[] = substr($class->name()->short(), 0, -4);
+        $subjectClassName = implode('\\', $subjectClassName);
+
+        try {
+            $subjectClass = $serviceLocator->reflector()->reflectClass($subjectClassName);
+        } catch (NotFound $e) {
+            $serviceLocator->logger()->warning(sprintf(
+                'Phpspec extension could not locate inferred class name "%s" '.
+                'for spec class "%s": %s',
+                $class->name()->full(),
+                $subjectClassName,
+                $e->getMessage()
+            ));
+            return VirtualReflectionMethodCollection::fromMembers([]);
+        }
+
+        $virtualMethods = [];
+        foreach ($subjectClass->methods() as $subjectMethod) {
+            $method = VirtualReflectionMethod::fromReflectionMethod($subjectMethod);
+            $subjectType = new ClassType(ClassName::fromString(self::SUBJECT_CLASS));
```

Comment:
could maybe add an intersection type here: `TypeFactory::intersection(TypeFactory::reflectedClassType($reflector, self::SUBJECT_CLASS), TypeFactory::reflectedClassType($reflector, ThePhpspecClass::class))`

(the types can be smarter if they have a Reflector instance in them)